docker/Dockerfile

```diff
@@ -1,27 +1,24 @@
 FROM alpine:edge AS builder
-RUN apk add --no-cache crystal shards libc-dev \
+RUN apk add --no-cache curl crystal shards libc-dev \
     yaml-dev libxml2-dev sqlite-dev zlib-dev openssl-dev \
-    sqlite-static zlib-static openssl-libs-static
+    yaml-static sqlite-static zlib-static openssl-libs-static
 WORKDIR /invidious
 COPY ./shard.yml ./shard.yml
-RUN shards update && shards install
-RUN apk add --no-cache curl && \
+RUN shards update && shards install && \
     curl -Lo /etc/apk/keys/omarroth.rsa.pub https://github.com/omarroth/boringssl-alpine/releases/download/1.1.0-r0/omarroth.rsa.pub && \
     curl -Lo boringssl-dev.apk https://github.com/omarroth/boringssl-alpine/releases/download/1.1.0-r0/boringssl-dev-1.1.0-r0.apk && \
     curl -Lo lsquic.apk https://github.com/omarroth/lsquic-alpine/releases/download/2.6.3-r0/lsquic-2.6.3-r0.apk && \
-    tar -xf boringssl-dev.apk && \
-    tar -xf lsquic.apk
-RUN mv ./usr/lib/libcrypto.a ./lib/lsquic/src/lsquic/ext/libcrypto.a && \
-    mv ./usr/lib/libssl.a ./lib/lsquic/src/lsquic/ext/libssl.a && \
-    mv ./usr/lib/liblsquic.a ./lib/lsquic/src/lsquic/ext/liblsquic.a
+    apk verify --no-cache boringssl-dev.apk lsquic.apk && \
+    tar -xf boringssl-dev.apk usr/lib/libcrypto.a usr/lib/libssl.a && \
+    tar -xf lsquic.apk usr/lib/liblsquic.a && \
+    mv usr/lib/* lib/lsquic/src/lsquic/ext && \
+    rm -r usr boringssl-dev.apk lsquic.apk /root/.cache
```

Comment:
Thanks for all of the work. Would it be possible to move these lines above `COPY ./shard.yml ./shard.yml` so that these commands can be run from cache even if `shard.yml` is changed?

Reply:
they actually overwrite the files shard generates. so if shard.yml changes we need to update and therefore need to overwrite them again.

Reply:
I'm sorry for nitpicking again, but what about

```patch
@@ -3,16 +3,19 @@ RUN apk add --no-cache curl crystal shards libc-dev \
     yaml-dev libxml2-dev sqlite-dev zlib-dev openssl-dev \
     yaml-static sqlite-static zlib-static openssl-libs-static
 WORKDIR /invidious
+RUN curl -Lo /etc/apk/keys/omarroth.rsa.pub https://github.com/omarroth/boringssl-alpine/releases/download/1.1.0-r0/omarroth.rsa.pub && \
+    curl -Lo ./boringssl-dev.apk https://github.com/omarroth/boringssl-alpine/releases/download/1.1.0-r0/boringssl-dev-1.1.0-r0.apk && \
+    curl -Lo ./lsquic.apk https://github.com/omarroth/lsquic-alpine/releases/download/2.6.3-r0/lsquic-2.6.3-r0.apk && \
+    apk verify --no-cache ./boringssl-dev.apk ./lsquic.apk && \
+    tar -xf ./boringssl-dev.apk ./usr/lib/libcrypto.a ./usr/lib/libssl.a && \
+    tar -xf ./lsquic.apk ./usr/lib/liblsquic.a && \
+    mkdir ./ext/ && \
+    mv ./usr/lib/* ./ext/ && \
+    rm -r ./usr/ /etc/apk/keys/omarroth.rsa.pub ./boringssl-dev.apk ./lsquic.apk /root/.cache
 COPY ./shard.yml ./shard.yml
 RUN shards update && shards install && \
-    curl -Lo /etc/apk/keys/omarroth.rsa.pub https://github.com/omarroth/boringssl-alpine/releases/download/1.1.0-r0/omarroth.rsa.pub && \
-    curl -Lo boringssl-dev.apk https://github.com/omarroth/boringssl-alpine/releases/download/1.1.0-r0/boringssl-dev-1.1.0-r0.apk && \
-    curl -Lo lsquic.apk https://github.com/omarroth/lsquic-alpine/releases/download/2.6.3-r0/lsquic-2.6.3-r0.apk && \
-    apk verify --no-cache boringssl-dev.apk lsquic.apk && \
-    tar -xf boringssl-dev.apk usr/lib/libcrypto.a usr/lib/libssl.a && \
-    tar -xf lsquic.apk usr/lib/liblsquic.a && \
-    mv usr/lib/* lib/lsquic/src/lsquic/ext && \
-    rm -r usr boringssl-dev.apk lsquic.apk /root/.cache
+    mv ./ext/* ./lib/lsquic/src/lsquic/ext/ && \
+    rm -r ./ext/
 COPY ./src/ ./src/
 # TODO: .git folder is required for building – this is destructive.
 # See definition of CURRENT_BRANCH, CURRENT_COMMIT and CURRENT_VERSION.
```

(untested). That way we don't need to `curl` again if `shared.yml` is changed. Also I updated to using `./` when targeting files in the current working directory for reasons of consistency.

Reply:
Oh, and what happened to `rm -rf /var/cache/apk/*`? I've never seen a cleanup of `/root/.cache` when working with `apk`.

Reply:
`--no-cache` and `rm -rf /var/cache/apk/*` has the same effect, `/root/.cache` is the share cache I found that doesn't need to be there, so I included it in the rm

Reply:
the change looks good tho, will add it in.